Update config schema

Toliak opened the pull request:
- Added reference link
- Added `buildMode` property
- Added default value into `luaLibImport`, `luaTarget`, `noImplicitSelf`, `noHeader` and `sourceMapTraceback` property
- Modified description of `luaPlugins` property
- Prettified

Toliak committed:
Fixes #4

- Added reference link
- Added `buildMode` property
- Added default value into `luaLibImport`, `luaTarget`, `noImplicitSelf`, `noHeader` and `sourceMapTraceback` property
- Modified description of `luaPlugins` property
- Prettified

Toliak commented:
This merge request will fix #4

lolleko commented on tsconfig-schema.json:
```diff
@@ -6,17 +6,35 @@
+              },
+              "import": {
+                "type": "string"
+              },
+              "after": {
+                "type": "boolean"
+              },
+              "afterDeclarations": {
+                "type": "boolean"
+              },
               "type": {
                 "oneOf": [
-                  { "const": "program" },
-                  { "const": "config" },
-                  { "const": "checker" },
-                  { "const": "raw" },
-                  { "const": "compilerOptions" }
+                  {
+                    "const": "program"
+                  },
+                  {
+                    "const": "config"
+                  },
+                  {
+                    "const": "checker"
+                  },
+                  {
+                    "const": "raw"
+                  },
+                  {
+                    "const": "compilerOptions"
+                  }
```
I don't think we actually need that part, at least not for VS Code support.
VS Code automatically merges our Schema with Typescript's.

Toliak replied:
Removed in a9440e48

Toliak committed:
In progress #5

- Added title and description for the schema
- Added (mixined) the default tsconfig.json schema
- Removed default properties (they are already described in the default tsconfig.json schema)

Toliak committed:
Fixed comma
